### PR TITLE
ADD: exception message to model (includes test cleanup and fixes)

### LINKED ISSUE #131: ExceptionHandler Implementation
Implement ExceptionHandler for Exception Handling

### LINKED ISSUE #159: Dataflow Violation
https://github.com/hhu-propra2/abschlussprojekt-mopse/blob/89355abed003d99f74059d0080d8743421c65f9a/src/main/java/mops/presentation/FileController.java#L102

Isn't that a dataflow violation. Wouldn't
```java
Directory dir;
```
do the trick?

### LINKED ISSUE #233: Remove unnecessary mocks from tests.

### PR DESCRIPTION
**Pull request by iTitus**
Closes #159 
Closes #131 
Closes #233

### CODE REVIEW

**Comment by iTitus** [comment]
Partially closes #131, still needs messages to be shown in the templates.

**Comment by Segelzwerg** [comment]
`http://localhost:8080/material1/dir/1` shows 500

**Comment by Segelzwerg** [comment]
> Shouldn't there be a handler for .../material1/

we could do redirect or throw an error but show html error is not so nice

**Comment by iTitus** [comment]
> 
> 
> `http://localhost:8080/material1/dir/1` shows 500

thats how it looks for me:
image

**Comment by Segelzwerg** [comment]
> > `http://localhost:8080/material1/dir/1` shows 500
> 
> thats how it looks for me:
> image

yes so was it for me when I worked on dummy data.

**Comment by Segelzwerg** [comment]
image

File: src/main/resources/templates/mops_error.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org"
+      th:replace="~{mopslayout :: html(name='Materialsammlung', title='Fehler', headcontent=~{:: headcontent}, navigation=~{:: navigation}, bodycontent=~{:: bodycontent})}">
+<head>
+    <meta charset="utf-8">
+    <title>Materialsammlung</title>
+    <th:block th:fragment="headcontent">
+        <!-- Links, Skripts, Styles hier einfügen! -->
```
Comment (Segelzwerg) [inline src/main/resources/templates/mops_error.html]: shouldn't this be removed?

File: src/main/resources/templates/mops_error.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    </th:block>
+</head>
+<body>
+<header>
+    <nav class="navigation navigation-secondary" is="mops-navigation" th:fragment="navigation">
+        <!-- Navigation als ungeordnete Liste mit einfachen Links hier einfügen! -->
```
Comment (Segelzwerg) [inline src/main/resources/templates/mops_error.html]: see above

File: src/main/resources/templates/mops_index.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org"
+      th:replace="~{mopslayout :: html(name='Materialsammlung', title='Index', headcontent=~{:: headcontent}, navigation=~{:: navigation}, bodycontent=~{:: bodycontent})}">
+<head>
+    <meta charset="utf-8">
+    <title>Materialsammlung</title>
+    <th:block th:fragment="headcontent">
+        <!-- Links, Skripts, Styles hier einfügen! -->
```
Comment (Segelzwerg) [inline src/main/resources/templates/mops_index.html]: above

File: src/main/resources/templates/mops_index.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    </th:block>
+</head>
+<body>
+<header>
+    <nav class="navigation navigation-secondary" is="mops-navigation" th:fragment="navigation">
+        <!-- Navigation als ungeordnete Liste mit einfachen Links hier einfügen! -->
```
Comment (Segelzwerg) [inline src/main/resources/templates/mops_index.html]: above

File: src/main/resources/templates/mops_index.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    </nav>
+</header>
+<main th:fragment="bodycontent">
+    <!-- Restlichen Inhalt hier einfügen! -->
```
Comment (Segelzwerg) [inline src/main/resources/templates/mops_index.html]: above

File: src/main/resources/templates/mops_error.html (hunk 1 above)
Comment (iTitus) [reply src/main/resources/templates/mops_error.html]: we could still add links, scripts etc. here. its basically a TODO

**Pull request merged**